src/parser/core/modules/Swiftcast.tsx

```diff
@@ -47,6 +47,25 @@ export abstract class SwiftcastModule extends BuffWindowModule {
 		severityTiers: this.severityTiers,
 	}
 
+	protected init() {
+		super.init()
+		// Inheriting the class doesn't update expectedGCDs's parameters when they
+		// override, so let's (re)define it here... feels mega jank tho
+		this.expectedGCDs = {
```

Comment:
Ah, yes, the joys of JS abstract classes.  Since I chose to have the expectedGCDs property be an object with the three properties on it for compactness, overriding individual object properties with "this.SomeOtherProperty" doesn't work unless you do it in init due to order of operations on class construction.

I'd say this raises a question of whether it's better to split up expectedGCDs (and probably the other three tracking objects) into separate properties rather than a consolidated object.  Alternatively, if you just have Swiftcast consumers override the full expectedGCD object (even at the cost of some code duplication), it won't require the init jank either.

Reply:
I guess I was thinking about how a user might implement this – if I were just trying to override to change only the warning message, I'd only want to set the one property instead of a whole object.

Reply:
Ah, forgot to update here as well. Per discord, I think we should just leave it in here so I can make the experience more straightforward to the users of the class – unless someone wants to block on me overriding `init`